spec/services/google/search_service_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Google::SearchService, type: :service do
+  let(:stubs) { Faraday::Adapter::Test::Stubs.new }
+  let(:stubbed_conn) { Faraday.new { |b| b.adapter(:test, stubs) } }
+
+  describe '#search!' do
+    before(:each) do
```

Comment:
Avoid using let and before blocks as much as possible: https://nimblehq.co/compass/development/code-conventions/ruby/rspec/#convention-15. There is added complexity when nesting context with the order of execution which can cause unexpected side effects. Prefer to inline the mock in each test even if it comes at the cost of repetitions, and define helper methods instead.

Reply:
Fixed in https://github.com/Goose97/google-scraper-ruby/pull/20/commits/a65b875f9864c4458482e66e048d38338ae16bad

However, this also makes some test cases exceed the spec example length limit. IMO, we can disable the lint rule in these cases